src/Microsoft.Framework.Runtime/DependencyManagement/LockFile.cs

```diff
@@ -15,18 +15,24 @@ public class LockFile
         public IList<LockFileLibrary> Libraries { get; set; } = new List<LockFileLibrary>();
         public IList<LockFileTarget> Targets { get; set; } = new List<LockFileTarget>();
 
-        public bool IsValidForProject(Project project)
+        public bool IsValidForProject(Project project, out string message)
         {
+            message = null;
+
             if (Version != LockFileReader.Version)
             {
+                message = $"Expected lock file version is {LockFileReader.Version}, while actual version is {Version}";
                 return false;
             }
 
             var actualTargetFrameworks = project.GetTargetFrameworks();
 
             // The lock file should contain dependencies for each framework plus dependencies shared by all frameworks
-            if (ProjectFileDependencyGroups.Count != actualTargetFrameworks.Count() + 1)
+            var frameworkNumDiff = ProjectFileDependencyGroups.Count - (actualTargetFrameworks.Count() + 1);
+            if (frameworkNumDiff != 0)
             {
+                var action = frameworkNumDiff < 0 ? "added to" : "removed from";
+                message = $"One or more frameworks were {action} {Project.ProjectFileName}";
```

Comment:
If we are going to localize this I believe this will have to be changed to 2 full strings anyways so not sure if using ternary just for a substring makes sense/difference...